pyincore_data/censusutil.py

```diff
@@ -38,15 +40,25 @@ def get_census_data(state: str = None, county: str = None, year: str = None, dat
                 data_name (str): Optional for getting different dataset. e.g, 'component'
 
             Returns:
-                dict, object: A json list and a dataframe for census api result
+                dict, obj, obj: A json list, a dataframe for census api result,
+                    and pyincore dataset
 
         """
         # create census api data url
         data_url = CensusUtil.generate_census_api_url(state, county, year, data_source, columns, geo_type, data_name)
 
         api_json, api_df = CensusUtil.request_census_api(data_url)
 
-        return api_json, api_df
+        # convert df to dataset
+        timestr = time.strftime("%Y%m%d-%H%M%S")
+        csv_name = 'api_' + str(timestr) + '.csv'
+        print('csv saved as ' + csv_name)
+        api_df.to_csv(csv_name)
+        out_dataset = Dataset.from_file(csv_name, data_type="ergo:censusdata")
```

Comment:
does this data_type `ergo:censusdata` already exist?

Reply:
yes it is already exists.